lib/hanami/cli/generators/app/component_context.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Hanami
+  module CLI
+    module Generators
+      module App
+        class ComponentContext < SliceContext
+          # Taken from lib/hanami/cli/generators/app/view_context.rb which mentions they should be moved somewhere. Not sure where yet.
+          MATCHER_PATTERN = /::|\./
+          private_constant :MATCHER_PATTERN
+
+          NAMESPACE_SEPARATOR = "::"
+          private_constant :NAMESPACE_SEPARATOR
+
+          INDENTATION = "  "
+          private_constant :INDENTATION
+
+          OFFSET = 1
+          private_constant :OFFSET
```

Comment:
For starters, let's try moving these to a generators/constants.rb, and make sure we have a manual require that ensures this file is loaded where necessary?

I'd be very happy to have this PR merge first, however, and then have you tackle that as a follow-up :)

Reply:
Yes, I think it should be done separately